log4j-api-test/src/test/java/org/apache/logging/log4j/message/ParameterFormatterTest.java

```diff
@@ -17,156 +17,167 @@
-        ParameterFormatter.formatMessage(sb, testMsg, args, 3);
-        final String result = sb.toString();
-        assertEquals("Test message ab c", result);
-    }
-
-    @Test
-    public void testFormatMessageNullArgs() {
-        final String testMsg = "Test message {} {} {} {} {} {}";
-        final String[] args = { "a", null, "c", null, null, null };
-        final StringBuilder sb = new StringBuilder();
-        ParameterFormatter.formatMessage(sb, testMsg, args, 6);
-        final String result = sb.toString();
-        assertEquals("Test message a null c null null null", result);
-    }
-
-    @Test
-    public void testFormatMessageStringArgsIgnoresSuperfluousArgs() {
-        final String testMsg = "Test message {}{} {}";
-        final String[] args = { "a", "b", "c", "unnecessary", "superfluous" };
-        final StringBuilder sb = new StringBuilder();
-        ParameterFormatter.formatMessage(sb, testMsg, args, 5);
-        final String result = sb.toString();
-        assertEquals("Test message ab c", result);
-    }
-
-    @Test
-    public void testFormatMessageStringArgsWithEscape() {
-        final String testMsg = "Test message \\{}{} {}";
-        final String[] args = { "a", "b", "c" };
-        final StringBuilder sb = new StringBuilder();
-        ParameterFormatter.formatMessage(sb, testMsg, args, 3);
-        final String result = sb.toString();
-        assertEquals("Test message {}a b", result);
-    }
-
-    @Test
-    public void testFormatMessageStringArgsWithTrailingEscape() {
-        final String testMsg = "Test message {}{} {}\\";
-        final String[] args = { "a", "b", "c" };
-        final StringBuilder sb = new StringBuilder();
-        ParameterFormatter.formatMessage(sb, testMsg, args, 3);
-        final String result = sb.toString();
-        assertEquals("Test message ab c\\", result);
-    }
-
-    @Test
-    public void testFormatMessageStringArgsWithTrailingEscapedEscape() {
-        final String testMsg = "Test message {}{} {}\\\\";
-        final String[] args = { "a", "b", "c" };
-        final StringBuilder sb = new StringBuilder();
-        ParameterFormatter.formatMessage(sb, testMsg, args, 3);
-        final String result = sb.toString();
-        assertEquals("Test message ab c\\\\", result);
-    }
-
-    @Test
-    public void testFormatMessageStringArgsWithEscapedEscape() {
-        final String testMsg = "Test message \\\\{}{} {}";
-        final String[] args = { "a", "b", "c" };
-        final StringBuilder sb = new StringBuilder();
-        ParameterFormatter.formatMessage(sb, testMsg, args, 3);
-        final String result = sb.toString();
-        assertEquals("Test message \\ab c", result);
+    @ParameterizedTest
+    @CsvSource({
+            "0,,false,",
+            "0,,false,aaa",
+            "0,,true,\\{}",
+            "1,0,false,{}",
+            "1,0,true,{}\\{}",
+            "1,2,true,\\\\{}",
+            "2,8:10,true,foo \\{} {}{}",
```

Comment:
@rgoers, these two scenarios are new.

Reply:
Maybe we could also test `{\\}` as an alternative to `\\{}`.

Reply:
@ppkarwasz, good point. Added in 763bc271b1bd3dbe4bed0361d2bbab6d688abb28.